eth2/types/src/validator.rs

```diff
@@ -81,6 +82,54 @@ impl Default for Validator {
     }
 }
 
+impl Decode for Validator {
+    fn is_ssz_fixed_len() -> bool {
+        true
+    }
+
+    fn ssz_fixed_len() -> usize {
+        <PublicKeyBytes as Decode>::ssz_fixed_len()
+            + <Hash256 as Decode>::ssz_fixed_len()
+            + <u64 as Decode>::ssz_fixed_len()
+            + <bool as Decode>::ssz_fixed_len()
+            + <Epoch as Decode>::ssz_fixed_len()
+            + <Epoch as Decode>::ssz_fixed_len()
+            + <Epoch as Decode>::ssz_fixed_len()
+            + <Epoch as Decode>::ssz_fixed_len()
+    }
+
+    fn from_ssz_bytes(bytes: &[u8]) -> Result<Self, DecodeError> {
+        if bytes.len() != <Self as Decode>::ssz_fixed_len() {
+            return Err(DecodeError::InvalidByteLength {
+                len: bytes.len(),
+                expected: <Self as Decode>::ssz_fixed_len(),
+            });
+        }
+
+        let mut start = 0;
```

Comment:
The 0 assigned here is what's causing the "never read" warning, `let mut start;` should work